### PR TITLE
GH-1219: Forgot Password Page

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/storefrontlib/src/lib/user/user.module.ts [hunk 1]
```diff
@@ -6,8 +6,13 @@ import { ReactiveFormsModule } from '@angular/forms';
 
 import { LoginModule } from './login/login.module';
 import { RegisterComponent } from './register/register.component';
-import { ResetPasswordComponent } from './reset-password/reset-password.component';
-import { UserModule, UrlTranslationModule } from '@spartacus/core';
+import { ForgotPasswordComponent } from './forgot-password/forgot-password.component';
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/user/user.module.ts]: Can we create "ForgotPasswordModule"  and in UserModule, we can just import it?

File: projects/storefrontlib/src/lib/cms-lib/address-book/address-card/address-card.component.spec.ts [hunk 2]
```diff
@@ -31,7 +31,9 @@ describe('AddressCardComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [AddressBookModule, StoreModule.forRoot({})]
+      declarations: [AddressCardComponent],
+      imports: [StoreModule.forRoot({})],
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/cms-lib/address-book/address-card/address-card.component.spec.ts]: We removed all "StoreModule" from storefrontlib. So, please remove this import.

File: projects/storefrontlib/src/lib/user/forgot-password/forgot-password.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      const message = fixture.debugElement.nativeElement.querySelector(
+        '.invalid-feedback'
+      );
+      expect(input.valid).toBeTruthy();
+      expect(message).toBeFalsy();
+      expect(submit.nativeElement.disabled).toBeFalsy();
+    });
+  });
+
+  it('should not display error message when user cusor is still in the field.', () => {
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/user/forgot-password/forgot-password.component.spec.ts]: typo: cursor :)

Comment (plabadie) [reply projects/storefrontlib/src/lib/user/forgot-password/forgot-password.component.spec.ts]: Thanks!

File: projects/storefrontlib/src/lib/cms-lib/address-book/address-card/address-card.component.spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/storefrontlib/src/lib/cms-lib/address-book/address-card/address-card.component.spec.ts]: Good catch.

File: projects/storefrontlib/src/lib/user/user.module.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/src/lib/user/user.module.ts]: I agree.  This would be a better and more modular design.

**Pull request merged**